src/app/qgisapp.cpp

```diff
@@ -5853,6 +5855,117 @@ void QgisApp::saveMapAsImage()
 
 } // saveMapAsImage
 
+void QgisApp::saveMapAsPdf()
+{
+  QList< QgsMapDecoration * > decorations;
+  QString activeDecorations;
+  Q_FOREACH ( QgsDecorationItem *decoration, mDecorationItems )
+  {
+    if ( decoration->enabled() )
+    {
+      decorations << decoration;
+      if ( activeDecorations.isEmpty() )
+        activeDecorations = decoration->name().toLower();
+      else
+        activeDecorations += QString( ", %1" ).arg( decoration->name().toLower() );
+    }
+  }
+
+  QgsMapSaveDialog dlg( this, mMapCanvas, activeDecorations, QgsMapSaveDialog::Pdf );
+  if ( !dlg.exec() )
+    return;
+
+  QgsSettings settings;
+  QString lastUsedDir = settings.value( QStringLiteral( "UI/lastSaveAsImageDir" ), QDir::homePath() ).toString();
+  QString myFileName = QFileDialog::getSaveFileName( this, tr( "Save map as" ), lastUsedDir, tr( "PDF Format" ) + " (*.pdf *.PDF)" );
+  if ( myFileName != QLatin1String( "" ) )
```

Comment:
!myFileName.isEmpty()

(Bonus points if you remove the "Fischer Price" style my prefix!)